reana_server/rest/workflows.py

```diff
@@ -365,57 +363,70 @@ def create_workflow(user):  # noqa
             Request failed. Not implemented.
     """
     try:
-        if request.json:
-            if "object_kind" in request.json:
-                (
-                    reana_spec_file,
-                    git_url,
-                    workflow_name,
-                    git_branch,
-                    git_commit_sha,
-                ) = _get_reana_yaml_from_gitlab(request.json, user.id_)
-                git_data = {
-                    "git_url": git_url,
-                    "git_branch": git_branch,
-                    "git_commit_sha": git_commit_sha,
-                }
-            else:
-                # validate against schema
-                git_data = {}
-                reana_spec_file = request.json
-                workflow_name = ""
-            workflow_engine = reana_spec_file["workflow"]["type"]
-        elif request.args.get("spec"):
-            return jsonify("Not implemented"), 501
-        else:
+        if not request.json:
             raise Exception(
-                "Either remote repository or a reana spec need to \
-            be provided"
+                "Either remote repository or a reana specification needs to be provided"
             )
 
-        if workflow_engine not in REANA_WORKFLOW_ENGINES:
-            raise Exception("Unknown workflow type.")
+        request_from_gitlab = "object_kind" in request.json
+        if request_from_gitlab:
+            (
+                reana_spec_file,
+                git_url,
+                workflow_name,
+                git_branch,
+                git_commit_sha,
+            ) = _get_reana_yaml_from_gitlab(request.json, user.id_)
+            git_data = {
+                "git_url": git_url,
+                "git_branch": git_branch,
+                "git_commit_sha": git_commit_sha,
+            }
+        elif request.args.get("spec"):
+            return jsonify("Not implemented"), 501
```

Comment:
*suggestion(non-blocking):* I don't think this code is used at all. Or prevents something from happening. It has been carried here for 5 years. Maybe we should delete it (in a separate commit)?

Reply:
I think we should keep it. We do have "spec" in create_workflow OpenAPI contract and it is obviously also exposed  to clients via `r-commons` and such:

```json
          {
            "description": "Remote repository which contains a valid REANA specification.",
            "in": "query",
            "name": "spec",
            "required": false,
            "type": "string"
          },
```

So, if some client decides to pass it along, I think it is safe to keep returning "Not implemented".

(IOW, we should not remove the implementation whilst keeping the interface. An alternative would be to remove it from OpenAPI everywhere, if it is never used, but then this would create unnecessary complications for us to handle incompatibilities with API versioning. It's simpler to keep both interface and implementation and return "Not implemented" in case somebody passes it along.)